src/imitation/algorithms/adversarial/common.py

```diff
@@ -444,29 +442,28 @@ def _get_log_policy_act_prob(
         Returns:
             A batch of log policy action probabilities.
         """
-        log_policy_act_prob, log_policy_act_prob_th = None, None
-        if isinstance(self.gen_algo.policy, policies.ActorCriticPolicy):
+        if isinstance(self.policy, policies.ActorCriticPolicy):
             # policies.ActorCriticPolicy has a concrete implementation of
             # evaluate_actions to generate log_policy_act_prob given obs and actions.
-            _, log_policy_act_prob_th, _ = self.gen_algo.policy.evaluate_actions(
+            _, log_policy_act_prob_th, _ = self.policy.evaluate_actions(
                 obs_th,
                 acts_th,
             )
-        elif isinstance(self.gen_algo.policy, sac_policies.SACPolicy):
-            gen_algo_actor = self.gen_algo.policy.actor
+        elif isinstance(self.policy, sac_policies.SACPolicy):
+            gen_algo_actor = self.policy.actor
             # generate log_policy_act_prob from SAC actor.
             mean_actions, log_std, _ = gen_algo_actor.get_action_dist_params(obs_th)
             distribution = gen_algo_actor.action_dist.proba_distribution(
                 mean_actions,
                 log_std,
             )
-            log_policy_act_prob_th = distribution.log_prob(acts_th)
-            # getting log probability of actions from a continuous distribution
-            # might result in NaN values.
-            nan_log_prob_idx = th.where(th.isnan(log_policy_act_prob_th))
-            log_policy_act_prob_th[nan_log_prob_idx] = LOG_PROB_CUTOFF
-        if log_policy_act_prob_th is not None:
-            log_policy_act_prob = log_policy_act_prob_th.detach().cpu().numpy()
+            # SAC applies a squashing function to bound the actions to a finite range
+            # `acts_th` need to be scaled accordingly before computing log prob.
+            scaled_acts_th = self.policy.scale_action(acts_th)
+            log_policy_act_prob_th = distribution.log_prob(scaled_acts_th)
+        else:
+            return None
+        log_policy_act_prob = log_policy_act_prob_th.detach().cpu().numpy()
```

Comment:
```suggestion
        return log_policy_act_prob_th.detach().cpu().numpy()
```
and delete next line

Reply:
check!